src/test/java/Homework21.java

```diff
@@ -0,0 +1,72 @@
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class Homework21 extends BaseTest {
+
+    String playlistname ="Homework21 Test";
+
+    @Test
+    public void renamePlaylist() throws InterruptedException {
+        login("simplyd2d@gmail.com", "te$t$tudent");
+        clickSubmit();
+        choosePlaylist();
+        rightClickPlaylistNam();
+        clickEditOption();
+        //contextClickChoosePlaylist();
+        enterPlaylistName();
+
+        Assert.assertTrue(doesPlaylistExist());
+    }
+
+    public void choosePlaylist() {
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("a[href='#!/playlist/40418']"))).click();
+    }
+
+    public void rightClickPlaylistNam() {
+        WebElement Playlist = driver.findElement(By.xpath("//a[normalize-space()='Homework21']"));
+        Actions actions = new Actions(driver);
+        actions.contextClick(Playlist).perform();
+        wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//li[normalize-space()='Edit']")));
+    }
+    public static void clickEditOption()  {
+        WebElement EditOption = driver.findElement(By.cssSelector("li[data-testid='playlist-context-menu-edit-40418']"));
+        EditOption.click();
+
+    }
+    //public void contextClickChoosePlaylist(){
+        //wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("li[data-testid='playlist-context-menu-edit-40418']")));
+       // WebElement playlistElement = driver.findElement(By.cssSelector(""));
+       // Actions action = new Actions(driver);
+       // action.contextClick(playlistElement).perform();
+       // wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector(""))).click();
+    //}
+
+    //public static void enterPlaylistName(){
+        //WebElement playlistInputField = driver.findElement(By.cssSelector(""));
+        //playlistInputField.sendKeys((Keys.chord(Keys.CONTROL, "a", Keys.BACK_SPACE)));
+        //playlistInputField.sendKeys(Homework21 Test);
+        //playlistInputField.sendKeys(Keys.ENTER);
+    //}
+    public static void enterPlaylistName() {
+        String playListName = "Homework21 New Playlist";
+        WebElement EditInput = driver.findElement(By.cssSelector("input[name='name']"));
+        Actions actions = new Actions(driver);
+        actions.doubleClick().perform();
+        EditInput.sendKeys((Keys.chord(Keys.DELETE)));
+        EditInput.sendKeys((Keys.chord(Keys.CONTROL, "a", Keys.BACK_SPACE)));
+        EditInput.sendKeys(playListName);
+        EditInput.sendKeys(Keys.ENTER);
+
+    }
+    public boolean doesPlaylistExist(){
+        WebElement playlistElement = driver.findElement(By.xpath("//a[normalize-space()='Homework21 New Playlist']"));
+        return playlistElement.isDisplayed();
+
+    }
+}
+
```

Comment:
The code is not compiled! Please finish the homework!You got this problem, please correct it. 

![image](https://user-images.githubusercontent.com/104761482/215751001-1944d370-dc2c-42dc-8d6f-47e38a0dd5ac.png)

Hint Please add @BeforeSuite in BaseTest

```
   @BeforeSuite
    static void setupClass() {
        WebDriverManager.chromedriver().setup();
    }

```

Reply:
Can you please explain a little more in detail ? I am getting a passing test but i don't understand why. I have in the Basetest @BeforeSuite above.
Thank you!

![Screen Shot 2023-01-30 at 7 17 50 PM](https://user-images.githubusercontent.com/121208284/215909270-08507010-d2c7-4cdd-8133-7edc0be0b17e.png)